base/string_utils.cpp

```diff
@@ -113,6 +113,21 @@ UniString Normalize(UniString const & s)
   return result;
 }
 
+void NormalizeDigits(string & utf8)
+{
+  for (size_t i = 0; i + 2 < utf8.size(); ++i)
+  {
+    if (utf8[i] == '\xEF' && utf8[i + 1] == '\xBC')
+    {
+      char const n = utf8[i + 2];
+      if (n < '\x90' || n > '\x99')
+        continue;
+      utf8[i] = n - 0x90 + '0';
+      utf8.erase(i + 1, 2);
```

Comment:
На практике чаще всего встречается одна цифра, изредка две. И чтобы оптимально написать твое решение, нужно заводить вторую функцию либо постоянно делать дополнительную проверку.

Если предложишь простое решение, с удовольствием его внедрю. 

> On 1.4.2016, at 22.47, ygorshenin <notifications@github.com> wrote:
> 
> In base/string_utils.cpp:
> 
> > @@ -113,6 +113,21 @@ UniString Normalize(UniString const & s)
> >    return result;
> >  }
> >  
> > +void NormalizeDigits(string & utf8)
> > +{
> > +  for (size_t i = 0; i + 2 < utf8.size(); ++i)
> > +  {
> > +    if (utf8[i] == '\xEF' && utf8[i + 1] == '\xBC')
> > +    {
> > +      char const n = utf8[i + 2];
> > +      if (n < '\x90' || n > '\x99')
> > +        continue;
> > +      utf8[i] = n - 0x90 + '0';
> > +      utf8.erase(i + 1, 2);
> Erase из середины строки выполняется за О от ее длины. А у тебя тут хоть n erase ов.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>